Implemented ejecting a floppy disk, which mimics a user clicking on the floppy drive eject button

TorZidan opened the pull request:
Implemented ejecting a floppy disk, which mimics a user clicking on the floppy drive eject button.

How it works:
If a floppy is already inserted and the user mouse-clicks in the floppy drive area of the lisa skin,
we pop a "Do you want to eject the diskette?" Yes/No dialog.
If yes, we trigger a  Floppy IRQ, the Lisa software intercepts it, and it sends a "floppy eject" command FLOP_CMD_UCLAMP, and we process it (in floppy.c) and eject the floppy.
I did not add a new "Eject Diskette" menu item, to keep things simple.

Why I chose to do this:
The PascalWorkshop1.0 installer asks me do eject floppy disk 2 from the lower drive and then insert disk 3. So I can't finish the installation without this.

Note 1: To eject the lower floppy in Lisa 1 mode, hold the "Shift" key and then click on the floppy drive (this is needed because the lisa skin shows only one floppy drive).
Note 2: Lisa may refuse to act on the IRQ, and hence refuse to eject the floppy. This is evident at the bios boot menu screen.
Note 3: There is also another way to eject a floppy: if we are booted into e.g. LOS3.1, we can eject a floppy using Lisa's menu.

In this commit I also removed the macro "LISA_SMALLTALK_RUNNING" which is not needed and is a leftover from my previous work on fixing the mouse cursor in Smalltalk.

busterswt commented:
@TorZidan - Can you verify that the mouse is still working within Smalltalk after the removal of [this code](https://github.com/arcanebyte/lisaem/commit/839765dd32a733630468360d4b00bef4257e1269)? In other words, is Smalltalk still functioning as-expected in current master branch? I got the impression it was only the logging message that was inaccurate, not the actual implementation. Which would mean we need to add that back.

> The only thing I would point out for commit https://github.com/arcanebyte/lisaem/commit/839765dd32a733630468360d4b00bef4257e1269
is that it **logs** that its for Smalltalk, when the values should be good for **any** Monitor 11 application using DRIVERS.OBJ for mouse handling (e.g. EDITOR.OBJ).

busterswt commented:
@TorZidan Do you happen to know what the eject experience would've been on an actual Lisa 1? Were there manual eject buttons for the Twiggy drives, and would the user have triggered an eject operation on that lower drive by pressing the button on the faceplate? I guess the guide calls this out:

> One disk-release button is
located above each drive
slot, to the right of the drive
number. When you want to
remove a diskette, press
the disk-release button and
wait for the Lisa to eject the
diskette. There may be a
delay while the computer
updates and releases the
diskette.

TorZidan commented:
[busterswt](https://github.com/busterswt), thanks for raising this question.

I did another round of testing of Smalltalk.
It turns out that my previous fix (and a similar fix proposed by pablo_marx@)  worked only in Monitor v11.8.

So I added more fixes in glue.c here, in this pull request, for Monitor 11.0 and 11.1 (it turns out that the mouse "vector" is different here from v11.8).

About your concerns about code that I have removed in my previous commit:
The code was there, I had merged it into the "else if" for Monitor 1, and it was working fine with Monitor v11.8.
We don't want to log it as "Lisa Smalltalk running", because we are in the Monitor operating system. So we log it as "Lisa Monitor running".

About the floppy eject buttons: yes, Lisa 1 has a floppy eject button next to each twiggy drive.
As I noted previously, it is up to the currently-running Lisa software to eject the floppy when  pressing the button: for example, in the Lisa bios boot screen, it refuses to eject a floppy when the button is pressed (it is not implemented in the bios code). 

Enjoy!
